mitmproxy/http.py

```diff
@@ -372,7 +372,8 @@ def set_content(self, value: Optional[bytes]) -> None:
             # Let's remove it!
             del self.headers["content-encoding"]
             self.raw_content = value
-        self.headers["content-length"] = str(len(self.raw_content))
+        if not "transfer-encoding" in self.headers:
```

Comment:
I'm still not sure if this should be: `if self.is_http2 or not "transfer-encoding" in self.headers:` since we probably still want to set `content-length` if (the unused/unsupported) `"transfer-encoding"` header is given in a HTTP/2 request. What do you think?

Reply:
HTTP/2 does not have `transfer-encoding` headers, so I think it's really good enough to do the check below.